### PR TITLE
Improve support for custom html-webpack-plugin templates

### PR DESCRIPTION
**Pull request by jscheid**

### CODE REVIEW

**Comment by jscheid** [comment]
@jantimon any chance I can get you to review this change, when you find some time?

File: index.js [hunk 1]
```diff
@@ -15,6 +15,17 @@ function findDepChunks(chunk, allDepChunkIds) {
   });
 }
 
+function htmlWebpackAssetSrc(compiler, htmlWebpackPlugin, src) {
+  return path.relative(
```
Comment (jantimon) [inline index.js]: This is not readable at all - what is happening in here?

**Comment by jantimon** [comment]
What exactly did you change?

**Comment by jscheid** [comment]
@jantimon thanks for looking at the code quality, I'll have a look at those issues.

Sorry, I didn't provide enough context:

This change adds the properties `jsIntegrity` and `cssIntegrity` on `htmlWebpackPlugin.files`, which enables giving people more straightforward and more complete instructions to follow for when they use HWP with `inject: false`. For example, the current instructions (on master) don't explain how to get the integrity value for stylesheets.

Perhaps instead of a review I should have asked if the new approach seems like one that you agree with, and one that is likely to keep working with future versions of html-webpack-plugin?

**Comment by jantimon** [comment]
The html-webpack-plugin did always focus on backwards compatibility - so if this works now it should also work in future ;) maybe we should add some integration tests also to the html webpack plugin

**Comment by jscheid** [comment]
Sounds good, so what would be the next step? Do you see any issues with merging this branch as it is now? I'm keen on getting 0.6.0 out the door soon.

File: index.js [hunk 2]
```diff
@@ -193,11 +216,30 @@ SubresourceIntegrityPlugin.prototype.apply = function apply(compiler) {
...
+      ['js', 'css'].forEach(function addIntegrity(fileType) {
+        pluginArgs.assets[fileType + 'Integrity'] =
+          pluginArgs.assets[fileType].map(function assetIntegrity(filePath) {
+            var src = hwpSrcRelativeOutputPath(compilation.compiler,
+                                               pluginArgs.plugin,
+                                               filePath);
+            return compilation.assets[src].integrity;
+          });
+      });
+      callback(null);
```
Comment (jantimon) [inline index.js]: This might cause a deprecated warning (not 100% sure) could you change it to `callback(null, pluginArgs)`?

**Comment by jantimon** [comment]
Well I think this change is a great improvement.
I also listed webpack-subresource-integrity in the html-webpack-plugin readme

**Comment by jscheid** [comment]
Thanks for your help with this @jantimon and for the listing in the readme. Released in v0.6.0.

**Pull request merged**